src/devices/Display/Matrix16x8.cs

```diff
@@ -0,0 +1,204 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+
+namespace Iot.Device.Display
+{
+    /// <summary>
+    /// Represents an 8x16 LED matrix using the HT16K33 LED Matrix driver.
+    /// </summary>
+    // Product: https://www.adafruit.com/product/2042
+    public class Matrix16x8 : Ht16k33
+    {
+        private byte[] _displayBuffer = new byte[17];
+
+        /// <summary>
+        /// Initialize Matrix display
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="I2cDevice"/> to create with.</param>
+        public Matrix16x8(I2cDevice i2cDevice)
+            : base(i2cDevice)
+        {
+        }
+
+        /// <summary>
+        /// Indexer for updating matrix.
+        /// </summary>
+        public int this[int x, int y]
+        {
+            set
+            {
+                UpdateBuffer(x, y, value);
+
+                if (BufferingEnabled)
+                {
+                    Flush();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Enable all LEDs.
+        /// </summary>
+        public void Fill()
+        {
+            _displayBuffer = new byte[]
+            {
+                0, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255
+            };
+
+            if (BufferingEnabled)
+            {
+                _i2cDevice.Write(_displayBuffer);
+            }
+        }
+
+        /// <inheritdoc/>
+        public override void Clear()
+        {
+            _displayBuffer = new byte[17];
```

Comment:
```suggestion
            _displayBuffer.AsSpan().Clear();
```
would avoid the allocation of the new array?
If it doesn't matter, keep the code as is.

Reply:
`_displayBuffer`-field could become `readonly` then.